### ISSUE TITLE
Invalid data type for min/max options. Must be string or number, but found 0.

### ISSUE THREAD

**Issue by jpm8888** [strapi/strapi#19482]
- Node.js version: 18.x.x
- NPM version: 10.2.3
- Strapi version: 4.20.0
- Database: Postgres 11
- Operating system: Linux
- Is your project Javascript or Typescript: Typescript

On running `strapi ts:generate-types --debug` getting this error.
`Failed to generate types for contentTypes: Invalid data type for min/max options. Must be string or number, but found 0. Exiting`

It was working on strapi version: 4.17.x

**Comment by Convly** [comment]
Hi, could you share the attribute definition that's failing? (the one with min or max being set to 0).

Is it coming from a manually generated content type schema? Or from one created using the content-type builder?

**Comment by Frost-on-Web** [comment]
Hi,
Got the issue at the moment from the "decimal" (and trying with "integer" is the same) type :
```
    "totalPrice": {
      "type": "decimal",
      "min": 0
    },
```

Must be modifed to a string to work :
```
    "totalPrice": {
      "type": "decimal",
      "min": "0"
    },
```

**Comment by Frost-on-Web** [comment]
Like the error said : 
`Must be string or number, but found 0.`

Probably an error of comparison.
Sorry to not have the time to look for into the code.

**Comment by Convly** [comment]
I've merged a fix that should be released tomorrow in 4.20.1.

Thanks for reporting this!
